Respawn/Checkpoint.cs

```diff
@@ -43,8 +45,34 @@ public class Checkpoint
                 await BuildDeleteTables(connection);
             }
 
+            if (_temporalTables.Count() > 0)

```

Comment:
You can use `Any()` instead of `Count()` since count will iterate all the items and `Any()` just checks for one item.